### PR TITLE
feat: Make Sipi handling optional during FileValue creation (Dev-2945)

### PR DESCRIPTION
**Pull request by mpro7**
### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [x] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [ ] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [x] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala [hunk 1]
```diff
@@ -2578,18 +2637,14 @@ case class FileValueWithSipiMetadata(fileValue: FileValueV2, sipiFileMetadata: F
  */
 object FileValueWithSipiMetadata {
   def fromJsonLdObject(
```
Comment (seakayone) [inline webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala:2639]: Definitely rename, maybe completely remove?

This is not getting the json ld object anymore. Not it is merely combining two constructors.

File: webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala [hunk 2]
```diff
@@ -1041,13 +1053,16 @@ object ValueContentV2 {
...
    * @param requestingUser       the user making the request.
    * @return a [[ValueContentV2]].
    */
   def fromJsonLdObject(
+    ingestState: AssetIngestState,
     jsonLdObject: JsonLDObject,
-    requestingUser: UserADM
+    requestingUser: UserADM,
+    shortcode: String
   ): ZIO[SipiService & StringFormatter & MessageRelay, Throwable, ValueContentV2] =
```
Comment (seakayone) [inline webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala:1066]: would be nice if we could remove the `SipiService` dependency here.

File: webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala [hunk 3]
```diff
@@ -73,12 +75,15 @@ final case class SipiServiceLive(
    * @return a [[FileMetadataSipiResponse]] containing the requested metadata.
    */
   override def getFileMetadata(filePath: String): Task[FileMetadataSipiResponse] =
```
Comment (seakayone) [inline webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala:77]: Would be nice to have named methods for`get from temp` and `get from permanent` and only provide the `internalFilename` as parameter.

File: webapi/src/test/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValuesV2Spec.scala [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      test("Expect file to be not ingested and in `tmp` folder") {
+        for {
+          info <- ValueContentV2
+                    .getFileInfo("0001", AssetIngestState.AssetInTemp, JsonLDUtil.parseJsonLD(json).body)
+                    .provide(ZLayer.succeed(new SipiService {
+                      override def getFileMetadata(filePath: String): Task[FileMetadataSipiResponse] =
+                        if (filePath.startsWith("/tmp")) ZIO.succeed(response)
+                        else ZIO.fail(AssertionException(filePath))
+
+                      override def moveTemporaryFileToPermanentStorage(
```
Comment (seakayone) [inline webapi/src/test/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValuesV2Spec.scala:52]: maybe extract to a configurable mock in this file. would make the test code more accessible.

**Review by seakayone — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala (hunk 3 above)
Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala:77]: done

**Pull request merged**